libp2p/NodeTable.cpp

```diff
@@ -104,7 +100,11 @@ void NodeTable::addNode(Node const& _node, NodeRelation _relation)
     auto ret = make_shared<NodeEntry>(m_hostNode.id, _node.id, _node.endpoint);
     DEV_GUARDED(x_nodes) { m_allNodes[_node.id] = ret; }
     LOG(m_logger) << "addNode pending for " << _node.id << "@" << _node.endpoint;
-    ping(_node.id, _node.endpoint);
+
+    // Avoid redundant ping - we don't want to ping if this addNode was invoked in response
+    // to the receipt of a ping packet
+    if (_ping)
+        ping(_node.id, _node.endpoint);
```

Comment:
Why do we need this ping anyway? The node will be checked in the following ping rounds, right?
I'd suggest to decouple ping from here.